File: crates/swiftlet/examples/gr_alternatives.rs
use std::sync::Arc;
use swiftlet::{ParserConfig, Swiftlet};

fn main() {
    let grammar = r#"
    start: hello | namaste
    hello: "Hello"
    namaste: "Namaste"
    "#;

    let texts = ["Hello", "Namaste"];

    let conf = Arc::new(ParserConfig::default());
    let parser = Swiftlet::from_str(grammar)
        .map(|grammar| grammar.parser(conf))
        .expect("failed to build parser");
    for text in texts {
        match parser.parse(&text) {
            Ok(ast) => {
                ast.print()
                // Output:
                // Tree("start", [Tree("hello", ["Hello"])])
                // Tree("start", [Tree("namaste", ["Namaste"])])
            }
            Err(err) => eprintln!("{}", err),
        }
    }
}

File: pyswiftlet/examples/gr_alternatives.py
"""Grammar alternatives example using the Swiftlet Python bindings."""

from swiftlet import Swiftlet


GRAMMAR = r"""
start: hello | namaste
hello: "Hello"
namaste: "Namaste"
"""


def main() -> None:
    parser = Swiftlet(GRAMMAR)
    for text in ("Hello", "Namaste"):
        tree = parser.parse(text)
        tree.pretty_print()


if __name__ == "__main__":
    main()
